base/src/main/java/io/spine/validate/FieldValidator.java

```diff
@@ -278,44 +273,53 @@ private boolean isRequiredId() {
      */
     private boolean isRequiredEntityId() {
         Required<V> requiredOption = Required.create(assumeRequired);
-        Optional<Boolean> requiredOptionValue = requiredOption.valueFrom(value);
+        Optional<Boolean> requiredOptionValue = requiredOption.valueFrom(descriptor());
         boolean notRequired = requiredOptionValue.isPresent() && !requiredOptionValue.get();
         return declaration.isEntityId() && !notRequired;
     }
 
-    private IfInvalidOption ifInvalid(FieldValue<V> fieldValue) {
-        IfInvalid<V> ifInvalidOption = new IfInvalid<>();
-        IfInvalidOption ifInvalid = ifInvalidOption.valueFrom(fieldValue)
+    private static IfInvalidOption ifInvalid(FieldDescriptor descriptor) {
+        IfInvalid ifInvalidOption = new IfInvalid();
+        IfInvalidOption ifInvalid = ifInvalidOption.valueFrom(descriptor)
                                                    .orElse(IfInvalidOption.getDefaultInstance());
         return ifInvalid;
     }
 
     private IfMissingOption ifMissing() {
-        IfMissing<V> ifMissing = new IfMissing<>();
-        return ifMissing.valueFrom(value)
+        IfMissing ifMissing = new IfMissing();
+        return ifMissing.valueFrom(descriptor())
```

Comment:
Please have a `valueOrDefault(...)` method and replace this duplicative `orElse` call.